src/OrbitGl/CallstackThreadBar.cpp

```diff
@@ -226,60 +226,117 @@ bool CallstackThreadBar::IsEmpty() const {
+  const std::string& function_name =
+      orbit_client_data::GetFunctionNameByAddress(*module_manager_, *capture_data_, address);
+  const std::string function_name_or_unknown_with_address =
+      (function_name == orbit_client_data::kUnknownFunctionOrModuleName)
+          ? absl::StrFormat("[unknown@%#x]", address)
           : function_name;
-  // Simple HTML escaping
-  return absl::StrReplaceAll(fn_name, {{"&", "&amp;"}, {"<", "&lt;"}, {">", "&gt;"}});
+
+  UnformattedModuleAndFunctionName module_and_function_name;
+  module_and_function_name.module_name = unknown_module_path_or_module_name;
+  module_and_function_name.module_is_unknown =
+      (module_path == orbit_client_data::kUnknownFunctionOrModuleName);
+  module_and_function_name.function_name = function_name_or_unknown_with_address;
+  module_and_function_name.function_is_unknown =
+      (function_name == orbit_client_data::kUnknownFunctionOrModuleName);
+  return module_and_function_name;
+}
+
+std::string CallstackThreadBar::FormatModuleName(
+    const CallstackThreadBar::UnformattedModuleAndFunctionName& module_and_function_name) {
+  return module_and_function_name.module_is_unknown
+             ? absl::StrFormat("<i>%s</i>", module_and_function_name.module_name)
+             : module_and_function_name.module_name;
 }
 
-constexpr const char* kErrorColorString = "#ffb000";
+std::string CallstackThreadBar::FormatFunctionName(
+    const CallstackThreadBar::UnformattedModuleAndFunctionName& module_and_function_name,
+    int max_length) {
+  const std::string& function_name = module_and_function_name.function_name;
+  const std::string shortened_function_name =
+      (max_length > 0) ? ShortenStringWithEllipsis(function_name, static_cast<size_t>(max_length))
+                       : function_name;
+  // Simple HTML escaping.
+  const std::string escaped_function_name =
+      absl::StrReplaceAll(shortened_function_name, {{"&", "&amp;"}, {"<", "&lt;"}, {">", "&gt;"}});
+  return module_and_function_name.function_is_unknown
+             ? absl::StrFormat("<i>%s</i>", escaped_function_name)
+             : escaped_function_name;
+}
 
-std::string CallstackThreadBar::FormatCallstackForTooltip(const CallstackInfo& callstack,
-                                                          int max_line_length, int max_lines,
-                                                          int bottom_n_lines) const {
-  std::string result;
-  int size = static_cast<int>(callstack.frames().size());
-  if (max_lines <= 0) {
-    max_lines = size;
-  }
-  const int bottom_n = std::min(std::min(max_lines - 1, bottom_n_lines), size);
-  const int top_n = std::min(max_lines, size) - bottom_n;
+constexpr const char* kUnwindErrorColorString = "#ffb000";
+
+std::string CallstackThreadBar::FormatCallstackForTooltip(const CallstackInfo& callstack) const {
+  constexpr int kMaxLineLength = 100;
+  constexpr int kMaxLines = 20;
+  constexpr int kBottomLineCount = 5;
```

Comment:
This part confused me a lot. What is this kBottomLineCount? I got now that it's because shortened for readability reasons. Consider to add a comment to explain this.

Reply:
This was kind of there before, I changed parameters into constants. Now I added a comment.